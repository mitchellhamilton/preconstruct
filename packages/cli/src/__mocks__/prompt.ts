import pLimit from "p-limit";

export let limit = pLimit(1);

export let promptInput = jest.fn();

export let doPromptInput = jest.fn();

export let createPromptConfirmLoader = () => jest.fn();
