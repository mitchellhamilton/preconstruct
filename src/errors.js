// @flow

/*::
import type { ItemUnion } from './types'
*/

export class FatalError extends Error {
  item: ItemUnion;
  constructor(message: string, item: ItemUnion) {
    super(message);
    this.item = item;
    if (process.env.NODE_ENV === "test") {
      let { getItemType } = require("./utils");
      this.message = `${this.message} in ${getItemType(item)} ${item.name}`;
    }
  }
}

export class ValidationError extends Error {}

export class MissingDependency extends Error {}
