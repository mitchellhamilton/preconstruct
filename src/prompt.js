// @flow
import inquirer from "inquirer";
import pLimit from "p-limit";
/*::
// this has to be in a comment because dependency import order reasons...
import { Package } from "./package";
*/
import DataLoader from "dataloader";
import chalk from "chalk";

export let limit = pLimit(1);

// there might be a simpler solution to this than using dataloader but it works so ¯\_(ツ)_/¯

type ItemUnion = Package;

let prefix = `🎁 ${chalk.green("?")}   `;

export function createPromptConfirmLoader(
  message: string
): (pkg: ItemUnion) => Promise<boolean> {
  let loader = new DataLoader<ItemUnion, boolean>(pkgs =>
    limit(
      () =>
        (async () => {
          if (pkgs.length === 1) {
            let { confirm } = await inquirer.prompt([
              {
                type: "confirm",
                name: "confirm",
                message,
                prefix: prefix + " " + pkgs[0].name
              }
            ]);
            return [confirm];
          }
          let { answers } = await inquirer.prompt([
            {
              type: "checkbox",
              name: "answers",
              message,
              choices: pkgs.map(pkg => ({ name: pkg.name, checked: true })),
              prefix
            }
          ]);
          return pkgs.map(pkg => {
            return answers.includes(pkg.name);
          });
        })(),
      { cache: false }
    )
  );

  return (pkg: ItemUnion) => loader.load(pkg);
}

let doPromptInput = async (
  message: string,
  pkg: ItemUnion,
  defaultAnswer?: string
): Promise<string> => {
  let { input } = await inquirer.prompt([
    {
      type: "input",
      name: "input",
      message,
      prefix: prefix + " " + pkg.name,
      default: defaultAnswer
    }
  ]);
  return input;
};

export let promptInput = (
  message: string,
  pkg: ItemUnion,
  defaultAnswer?: string
) => limit(() => doPromptInput(message, pkg, defaultAnswer));
