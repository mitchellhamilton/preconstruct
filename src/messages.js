// @flow

export let errors = {
  noEntryPoint:
    "no entrypoint was provided, please create a file at src/index.js",
  deniedWriteMainField:
    "changing the main field is required for preconstruct to build",
  invalidModuleField: "module field is invalid",
  invalidMainField: "main field is invalid",
  invalidUmdMainField: "umd:main field is invalid",
  invalidBrowserField: "browser field is invalid",
  invalidReactNativeField: "react-native field is invalid",
  umdNameNotSpecified:
    "the umd:main field is specified but a umdName option is not specified. please add it to the preconstruct field in your package.json",
  deniedWriteBrowserField:
    "building browser bundles for modules that include typeof window or typeof document is currently required"
};

import { createPromptConfirmLoader } from "./prompt";

export let confirms = {
  writeMainField: createPromptConfirmLoader(
    "preconstruct is going to change the main field in your package.json, are you okay with that?"
  ),
  writeModuleField: createPromptConfirmLoader(
    "would you like to generate module builds? this will write to the module field in your package.json"
  ),
  fixModuleField: createPromptConfirmLoader(
    "would you like to fix the module field?"
  ),
  writeUmdBuilds: createPromptConfirmLoader(
    "would you like to generate UMD builds?"
  ),
  shouldInstallObjectAssign: createPromptConfirmLoader(
    "Object.assign is polyfilled with object-assign to reduce bundle size when used with react. would you like to install object-assign automatically?"
  ),
  addBrowserField: createPromptConfirmLoader(
    "typeof window or typeof document is used in this package. would you like build seperate browser builds for better browser bundle sizes?"
  )
};

export let inputs = {
  getUmdName: "what should the name used for UMD bundles be?"
};

export let infos = {
  validMainField: "main field is valid",
  validModuleField: "module field is valid",
  validUmdMainField: "umd:main field is valid",
  validEntrypoint: "a valid entry point exists.",
  validBrowserField: "browser field is valid",
  validReactNativeField: "react-native field is valid"
};

export let successes = {
  validPackage: "package is valid!",
  startedWatching: "started watching!"
};
