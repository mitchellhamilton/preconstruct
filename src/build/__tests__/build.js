// @flow
import build from "../";
import fixturez from "fixturez";
import path from "path";
import spawn from "spawndamnit";
import { initBasic, getPkg, snapshotDistFiles } from "../../../test-utils";
import { confirms } from "../../messages";
import { FatalError } from "../../errors";
import { promptInput } from "../../prompt";

const f = fixturez(__dirname);

jest.mock("../../prompt");

let unsafePromptInput: any = promptInput;

async function install(tmpPath) {
  await spawn("yarn", ["install"], { cwd: tmpPath });
}

jest.mock("install-packages");

let unsafeRequire = require;

test("basic", async () => {
  let tmpPath = f.copy("valid-package");

  await build(tmpPath);

  await snapshotDistFiles(tmpPath);

  expect(unsafeRequire(tmpPath).default).toBe("something");
});

test("monorepo", async () => {
  let tmpPath = f.copy("monorepo");
  await initBasic(tmpPath);
  await install(tmpPath);
  await build(tmpPath);
  let counter = 1;
  for (let pkg of ["package-one", "package-two"]) {
    let pkgPath = path.join(tmpPath, "packages", pkg);
    await snapshotDistFiles(pkgPath);

    expect(unsafeRequire(pkgPath).default).toBe(counter++);
  }
});

test("no module", async () => {
  let tmpPath = f.copy("no-module");

  await build(tmpPath);

  await snapshotDistFiles(tmpPath);

  expect(unsafeRequire(tmpPath).default).toBe(
    "this does not have a module build"
  );
});

test.skip("uses obj spread", async () => {
  let tmpPath = f.copy("use-obj-spread");
  confirms.shouldInstallObjectAssign.mockReturnValue(Promise.resolve(false));

  try {
    await build(tmpPath);
  } catch (err) {
    expect(err).toBeInstanceOf(FatalError);
    expect(err.message).toBe(
      "object-assign should be in dependencies of use-object-spread"
    );
  }
});

test("clears dist folder", async () => {
  let tmpPath = f.copy("already-has-things-in-dist");

  await build(tmpPath);

  await snapshotDistFiles(tmpPath);
});

test("flow", async () => {
  let tmpPath = f.copy("flow");

  await build(tmpPath);

  await snapshotDistFiles(tmpPath);
});

test("flow", async () => {
  let tmpPath = f.copy("flow-export-default");

  await build(tmpPath);

  await snapshotDistFiles(tmpPath);
});

test("prod checks", async () => {
  let tmpPath = f.copy("prod-checks");

  await build(tmpPath);

  await snapshotDistFiles(tmpPath);
});

// TODO: make it faster so this isn't required
jest.setTimeout(20000);

test("browser", async () => {
  let tmpPath = f.copy("browser");

  confirms.addBrowserField.mockReturnValue(Promise.resolve(true));

  await build(tmpPath);

  await snapshotDistFiles(tmpPath);
});

test("umd with dep on other module", async () => {
  let tmpPath = f.copy("umd-with-dep");

  await install(tmpPath);

  unsafePromptInput.mockImplementation(async question => {
    if (question === `What should the umdName of react be?`) {
      return "React";
    }
    throw new Error("unexpected question: " + question);
  });

  await build(tmpPath);

  await snapshotDistFiles(tmpPath);
  expect(await getPkg(tmpPath)).toMatchInlineSnapshot(`
Object {
  "devDependencies": Object {
    "react": "^16.6.3",
  },
  "license": "MIT",
  "main": "dist/umd-with-dep.cjs.js",
  "name": "umd-with-dep",
  "peerDependencies": Object {
    "react": "^16.6.3",
  },
  "preconstruct": Object {
    "globals": Object {
      "react": "React",
    },
    "umdName": "umdWithDep",
  },
  "private": true,
  "umd:main": "dist/umd-with-dep.umd.min.js",
  "version": "1.0.0",
}
`);
});

test("monorepo umd with dep on other module", async () => {
  let tmpPath = f.copy("monorepo-umd-with-dep");
  let asked = false;
  unsafePromptInput.mockImplementation(async question => {
    if (asked) {
      throw new Error("only one prompt should happen: " + question);
    }
    if (question === `What should the umdName of react be?`) {
      asked = true;
      return "React";
    }
    throw new Error("unexpected question: " + question);
  });
  await install(tmpPath);
  await build(tmpPath);

  await snapshotDistFiles(path.join(tmpPath, "packages", "package-one"));
  await snapshotDistFiles(path.join(tmpPath, "packages", "package-two"));
  await snapshotDistFiles(path.join(tmpPath, "packages", "package-three"));
  await snapshotDistFiles(path.join(tmpPath, "packages", "package-four"));

  expect(await getPkg(path.join(tmpPath, "packages", "package-one")))
    .toMatchInlineSnapshot(`
Object {
  "devDependencies": Object {
    "react": "^16.6.3",
  },
  "license": "MIT",
  "main": "dist/package-one-umd-with-dep.cjs.js",
  "name": "@some-scope/package-one-umd-with-dep",
  "peerDependencies": Object {
    "react": "^16.6.3",
  },
  "preconstruct": Object {
    "umdName": "packageOne",
  },
  "private": true,
  "umd:main": "dist/package-one-umd-with-dep.umd.min.js",
  "version": "1.0.0",
}
`);

  expect(await getPkg(path.join(tmpPath, "packages", "package-two")))
    .toMatchInlineSnapshot(`
Object {
  "devDependencies": Object {
    "react": "^16.6.3",
  },
  "license": "MIT",
  "main": "dist/package-two-umd-with-dep.cjs.js",
  "name": "@some-scope/package-two-umd-with-dep",
  "peerDependencies": Object {
    "react": "^16.6.3",
  },
  "preconstruct": Object {
    "umdName": "packageTwo",
  },
  "private": true,
  "umd:main": "dist/package-two-umd-with-dep.umd.min.js",
  "version": "1.0.0",
}
`);

  expect(await getPkg(tmpPath)).toMatchInlineSnapshot(`
Object {
  "license": "MIT",
  "main": "index.js",
  "name": "monorepo-umd-with-dep",
  "preconstruct": Object {
    "globals": Object {
      "react": "React",
    },
    "packages": Array [
      "packages/*",
    ],
  },
  "private": true,
  "version": "1.0.0",
  "workspaces": Array [
    "packages/*",
  ],
}
`);
});

test.skip("uses @babel/runtime", async () => {
  let tmpPath = f.copy("use-babel-runtime");

  confirms.shouldInstallBabelRuntime.mockReturnValue(Promise.resolve(false));

  try {
    await build(tmpPath);
  } catch (err) {
    expect(err).toBeInstanceOf(FatalError);
    expect(err.message).toMatchInlineSnapshot(
      `"@babel/runtime should be in dependencies of use-babel-runtime"`
    );
  }
});

test("@babel/runtime installed", async () => {
  let tmpPath = f.copy("babel-runtime-installed");

  await install(tmpPath);

  await build(tmpPath);

  await snapshotDistFiles(tmpPath);
});

test.skip("@babel/runtime", async () => {
  let tmpPath = f.copy("babel-runtime-custom-babel");

  await install(tmpPath);

  await build(tmpPath);

  await snapshotDistFiles(tmpPath);
});

test("monorepo single package", async () => {
  let tmpPath = f.copy("monorepo-single-package");
  await initBasic(tmpPath);
  await install(tmpPath);

  await build(tmpPath);
  let pkgPath = path.join(tmpPath, "packages", "package-two");
  await snapshotDistFiles(pkgPath);

  expect(unsafeRequire(pkgPath).default).toBe(2);
});
