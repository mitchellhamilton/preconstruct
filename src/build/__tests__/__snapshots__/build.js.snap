// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@babel/runtime and object-assign: babel-runtime-and-object-assign.cjs.dev.js 1`] = `
"'use strict';

Object.defineProperty(exports, '__esModule', {
  value: true
});

function _interopDefault(ex) {
  return ex && typeof ex === 'object' && 'default' in ex ? ex['default'] : ex;
}

var _objectAssign = _interopDefault(require('object-assign'));

let something = _objectAssign({}, {
  something: true
});

class Bar {}

class Foo extends Bar {}

exports.something = something;
exports.Foo = Foo;
"
`;

exports[`@babel/runtime and object-assign: babel-runtime-and-object-assign.cjs.js 1`] = `
"'use strict';

if (process.env.NODE_ENV === \\"production\\") {
  module.exports = require(\\"./babel-runtime-and-object-assign.cjs.prod.js\\");
} else {
  module.exports = require(\\"./babel-runtime-and-object-assign.cjs.dev.js\\");
}
"
`;

exports[`@babel/runtime and object-assign: babel-runtime-and-object-assign.cjs.prod.js 1`] = `
"\\"use strict\\";
function _interopDefault(ex) {
  return ex && \\"object\\" == typeof ex && \\"default\\" in ex ? ex.default : ex;
}
Object.defineProperty(exports, \\"__esModule\\", { value: !0 });
var _objectAssign = _interopDefault(require(\\"object-assign\\"));
let something = _objectAssign({}, { something: !0 });
class Bar {}
class Foo extends Bar {}
(exports.something = something), (exports.Foo = Foo);
"
`;

exports[`@babel/runtime installed: babel-runtime-installed.cjs.dev.js 1`] = `
"'use strict';

Object.defineProperty(exports, '__esModule', {
  value: true
});

class Bar {}

class Foo extends Bar {}

exports.Foo = Foo;
"
`;

exports[`@babel/runtime installed: babel-runtime-installed.cjs.js 1`] = `
"'use strict';

if (process.env.NODE_ENV === \\"production\\") {
  module.exports = require(\\"./babel-runtime-installed.cjs.prod.js\\");
} else {
  module.exports = require(\\"./babel-runtime-installed.cjs.dev.js\\");
}
"
`;

exports[`@babel/runtime installed: babel-runtime-installed.cjs.prod.js 1`] = `
"\\"use strict\\";
Object.defineProperty(exports, \\"__esModule\\", { value: !0 });
class Bar {}
class Foo extends Bar {}
exports.Foo = Foo;
"
`;

exports[`@babel/runtime installed: babel-runtime-installed.esm.js 1`] = `
"class Bar {}

class Foo extends Bar {}

export { Foo };
"
`;

exports[`basic: valid-package.cjs.dev.js 1`] = `
"'use strict';

Object.defineProperty(exports, '__esModule', {
  value: true
});
var index = \\"something\\";
exports.default = index;
"
`;

exports[`basic: valid-package.cjs.js 1`] = `
"'use strict';

if (process.env.NODE_ENV === \\"production\\") {
  module.exports = require(\\"./valid-package.cjs.prod.js\\");
} else {
  module.exports = require(\\"./valid-package.cjs.dev.js\\");
}
"
`;

exports[`basic: valid-package.cjs.prod.js 1`] = `
"\\"use strict\\";
Object.defineProperty(exports, \\"__esModule\\", { value: !0 });
var index = \\"something\\";
exports.default = index;
"
`;

exports[`basic: valid-package.esm.js 1`] = `
"var index = \\"something\\";

export default index;
"
`;

exports[`basic: valid-package.umd.min.js 1`] = `
"!function(e,t){\\"object\\"==typeof exports&&\\"undefined\\"!=typeof module?module.exports=t():\\"function\\"==typeof define&&define.amd?define(t):(e=e||self).validPackage=t()}(this,function(){\\"use strict\\";return\\"something\\"});
//# sourceMappingURL=valid-package.umd.min.js.map
"
`;

exports[`basic: valid-package.umd.min.js.map 1`] = `"{\\"version\\":3,\\"file\\":\\"valid-package.umd.min.js\\",\\"sources\\":[\\"../src/index.js\\"],\\"sourcesContent\\":[\\"export default \\\\\\"something\\\\\\";\\\\n\\"],\\"names\\":[],\\"mappings\\":\\"wMAAe\\"}"`;

exports[`browser: valid-package.browser.cjs.js 1`] = `
"'use strict';

Object.defineProperty(exports, '__esModule', {
  value: true
});
let thing = \\"wow\\";
{
  thing = \\"something\\";
}
{
  thing += \\"other\\";
}
var thing$1 = thing;
exports.default = thing$1;
"
`;

exports[`browser: valid-package.browser.esm.js 1`] = `
"let thing = \\"wow\\";

{
  thing = \\"something\\";
}

{
  thing += \\"other\\";
}

var thing$1 = thing;

export default thing$1;
"
`;

exports[`browser: valid-package.cjs.dev.js 1`] = `
"'use strict';

Object.defineProperty(exports, '__esModule', {
  value: true
});
let thing = \\"wow\\";

if (typeof window !== \\"undefined\\") {
  thing = \\"something\\";
}

if (typeof document !== undefined) {
  thing += \\"other\\";
}

var thing$1 = thing;
exports.default = thing$1;
"
`;

exports[`browser: valid-package.cjs.js 1`] = `
"'use strict';

if (process.env.NODE_ENV === \\"production\\") {
  module.exports = require(\\"./valid-package.cjs.prod.js\\");
} else {
  module.exports = require(\\"./valid-package.cjs.dev.js\\");
}
"
`;

exports[`browser: valid-package.cjs.prod.js 1`] = `
"\\"use strict\\";
Object.defineProperty(exports, \\"__esModule\\", { value: !0 });
let thing = \\"wow\\";
\\"undefined\\" != typeof window && (thing = \\"something\\"),
  void 0 !== typeof document && (thing += \\"other\\");
var thing$1 = thing;
exports.default = thing$1;
"
`;

exports[`browser: valid-package.esm.js 1`] = `
"let thing = \\"wow\\";

if (typeof window !== \\"undefined\\") {
  thing = \\"something\\";
}

if (typeof document !== undefined) {
  thing += \\"other\\";
}

var thing$1 = thing;

export default thing$1;
"
`;

exports[`clears dist folder: already-has-things-in-dist.cjs.dev.js 1`] = `
"'use strict';

Object.defineProperty(exports, '__esModule', {
  value: true
});
var index = \\"something\\";
exports.default = index;
"
`;

exports[`clears dist folder: already-has-things-in-dist.cjs.js 1`] = `
"'use strict';

if (process.env.NODE_ENV === \\"production\\") {
  module.exports = require(\\"./already-has-things-in-dist.cjs.prod.js\\");
} else {
  module.exports = require(\\"./already-has-things-in-dist.cjs.dev.js\\");
}
"
`;

exports[`clears dist folder: already-has-things-in-dist.cjs.prod.js 1`] = `
"\\"use strict\\";
Object.defineProperty(exports, \\"__esModule\\", { value: !0 });
var index = \\"something\\";
exports.default = index;
"
`;

exports[`flow: flow.cjs.dev.js 1`] = `
"'use strict';

Object.defineProperty(exports, '__esModule', {
  value: true
});
var a = \\"wow\\";

function doSomething(arg) {
  return \\"something\\" + arg;
}

exports.doSomething = doSomething;
exports.something = a;
"
`;

exports[`flow: flow.cjs.js 1`] = `
"'use strict';

if (process.env.NODE_ENV === \\"production\\") {
  module.exports = require(\\"./flow.cjs.prod.js\\");
} else {
  module.exports = require(\\"./flow.cjs.dev.js\\");
}
"
`;

exports[`flow: flow.cjs.js.flow 1`] = `
"// @flow
export * from \\"../src/index.js\\";
"
`;

exports[`flow: flow.cjs.prod.js 1`] = `
"\\"use strict\\";
Object.defineProperty(exports, \\"__esModule\\", { value: !0 });
var a = \\"wow\\";
function doSomething(arg) {
  return \\"something\\" + arg;
}
(exports.doSomething = doSomething), (exports.something = a);
"
`;

exports[`flow: flow.esm.js 1`] = `
"var a = \\"wow\\";

function doSomething(arg) {
  return \\"something\\" + arg;
}

export { doSomething, a as something };
"
`;

exports[`flow: flow-export-default.cjs.dev.js 1`] = `
"'use strict';

Object.defineProperty(exports, '__esModule', {
  value: true
});

function doSomething(arg) {
  return \\"something\\" + arg;
}

var index = \\"wow\\";
exports.doSomething = doSomething;
exports.default = index;
"
`;

exports[`flow: flow-export-default.cjs.js 1`] = `
"'use strict';

if (process.env.NODE_ENV === \\"production\\") {
  module.exports = require(\\"./flow-export-default.cjs.prod.js\\");
} else {
  module.exports = require(\\"./flow-export-default.cjs.dev.js\\");
}
"
`;

exports[`flow: flow-export-default.cjs.js.flow 1`] = `
"// @flow
export * from \\"../src/index.js\\";
export { default } from \\"../src/index.js\\";
"
`;

exports[`flow: flow-export-default.cjs.prod.js 1`] = `
"\\"use strict\\";
function doSomething(arg) {
  return \\"something\\" + arg;
}
Object.defineProperty(exports, \\"__esModule\\", { value: !0 });
var index = \\"wow\\";
(exports.doSomething = doSomething), (exports.default = index);
"
`;

exports[`flow: flow-export-default.esm.js 1`] = `
"function doSomething(arg) {
  return \\"something\\" + arg;
}
var index = \\"wow\\";

export default index;
export { doSomething };
"
`;

exports[`monorepo umd with dep on other module: package-four-umd-with-dep.cjs.dev.js 1`] = `
"'use strict';

require('@some-scope/package-one-umd-with-dep');
"
`;

exports[`monorepo umd with dep on other module: package-four-umd-with-dep.cjs.js 1`] = `
"'use strict';

if (process.env.NODE_ENV === \\"production\\") {
  module.exports = require(\\"./package-four-umd-with-dep.cjs.prod.js\\");
} else {
  module.exports = require(\\"./package-four-umd-with-dep.cjs.dev.js\\");
}
"
`;

exports[`monorepo umd with dep on other module: package-four-umd-with-dep.cjs.prod.js 1`] = `
"\\"use strict\\";
require(\\"@some-scope/package-one-umd-with-dep\\");
"
`;

exports[`monorepo umd with dep on other module: package-four-umd-with-dep.umd.min.js 1`] = `
"!function(e,t){\\"object\\"==typeof exports&&\\"undefined\\"!=typeof module?t(require(\\"react\\")):\\"function\\"==typeof define&&define.amd?define([\\"react\\"],t):t((e=e||self).React)}(this,function(e){\\"use strict\\";e.createElement(\\"div\\",null)});
//# sourceMappingURL=package-four-umd-with-dep.umd.min.js.map
"
`;

exports[`monorepo umd with dep on other module: package-four-umd-with-dep.umd.min.js.map 1`] = `"{\\"version\\":3,\\"file\\":\\"package-four-umd-with-dep.umd.min.js\\",\\"sources\\":[\\"../../package-one/src/index.js\\"],\\"sourcesContent\\":[\\"import { createElement } from \\\\\\"react\\\\\\";\\\\n\\\\ncreateElement(\\\\\\"div\\\\\\", null);\\\\n\\"],\\"names\\":[],\\"mappings\\":\\"sNAEc,MAAO\\"}"`;

exports[`monorepo umd with dep on other module: package-one-umd-with-dep.cjs.dev.js 1`] = `
"'use strict';

var react = require('react');

react.createElement(\\"div\\", null);
"
`;

exports[`monorepo umd with dep on other module: package-one-umd-with-dep.cjs.js 1`] = `
"'use strict';

if (process.env.NODE_ENV === \\"production\\") {
  module.exports = require(\\"./package-one-umd-with-dep.cjs.prod.js\\");
} else {
  module.exports = require(\\"./package-one-umd-with-dep.cjs.dev.js\\");
}
"
`;

exports[`monorepo umd with dep on other module: package-one-umd-with-dep.cjs.prod.js 1`] = `
"\\"use strict\\";
var react = require(\\"react\\");
react.createElement(\\"div\\", null);
"
`;

exports[`monorepo umd with dep on other module: package-one-umd-with-dep.umd.min.js 1`] = `
"!function(e,t){\\"object\\"==typeof exports&&\\"undefined\\"!=typeof module?t(require(\\"react\\")):\\"function\\"==typeof define&&define.amd?define([\\"react\\"],t):t((e=e||self).React)}(this,function(e){\\"use strict\\";e.createElement(\\"div\\",null)});
//# sourceMappingURL=package-one-umd-with-dep.umd.min.js.map
"
`;

exports[`monorepo umd with dep on other module: package-one-umd-with-dep.umd.min.js.map 1`] = `"{\\"version\\":3,\\"file\\":\\"package-one-umd-with-dep.umd.min.js\\",\\"sources\\":[\\"../src/index.js\\"],\\"sourcesContent\\":[\\"import { createElement } from \\\\\\"react\\\\\\";\\\\n\\\\ncreateElement(\\\\\\"div\\\\\\", null);\\\\n\\"],\\"names\\":[],\\"mappings\\":\\"sNAEc,MAAO\\"}"`;

exports[`monorepo umd with dep on other module: package-three-umd-with-dep.cjs.dev.js 1`] = `
"'use strict';

require('@some-scope/package-one-umd-with-dep');
"
`;

exports[`monorepo umd with dep on other module: package-three-umd-with-dep.cjs.js 1`] = `
"'use strict';

if (process.env.NODE_ENV === \\"production\\") {
  module.exports = require(\\"./package-three-umd-with-dep.cjs.prod.js\\");
} else {
  module.exports = require(\\"./package-three-umd-with-dep.cjs.dev.js\\");
}
"
`;

exports[`monorepo umd with dep on other module: package-three-umd-with-dep.cjs.prod.js 1`] = `
"\\"use strict\\";
require(\\"@some-scope/package-one-umd-with-dep\\");
"
`;

exports[`monorepo umd with dep on other module: package-three-umd-with-dep.umd.min.js 1`] = `
"!function(e){\\"function\\"==typeof define&&define.amd?define([\\"@some-scope/package-one-umd-with-dep\\"],e):e()}(function(){});
//# sourceMappingURL=package-three-umd-with-dep.umd.min.js.map
"
`;

exports[`monorepo umd with dep on other module: package-three-umd-with-dep.umd.min.js.map 1`] = `"{\\"version\\":3,\\"file\\":\\"package-three-umd-with-dep.umd.min.js\\",\\"sources\\":[],\\"sourcesContent\\":[],\\"names\\":[],\\"mappings\\":\\"\\"}"`;

exports[`monorepo umd with dep on other module: package-two-umd-with-dep.cjs.dev.js 1`] = `
"'use strict';

var react = require('react');

react.createElement(\\"h1\\", null);
"
`;

exports[`monorepo umd with dep on other module: package-two-umd-with-dep.cjs.js 1`] = `
"'use strict';

if (process.env.NODE_ENV === \\"production\\") {
  module.exports = require(\\"./package-two-umd-with-dep.cjs.prod.js\\");
} else {
  module.exports = require(\\"./package-two-umd-with-dep.cjs.dev.js\\");
}
"
`;

exports[`monorepo umd with dep on other module: package-two-umd-with-dep.cjs.prod.js 1`] = `
"\\"use strict\\";
var react = require(\\"react\\");
react.createElement(\\"h1\\", null);
"
`;

exports[`monorepo umd with dep on other module: package-two-umd-with-dep.umd.min.js 1`] = `
"!function(e,t){\\"object\\"==typeof exports&&\\"undefined\\"!=typeof module?t(require(\\"react\\")):\\"function\\"==typeof define&&define.amd?define([\\"react\\"],t):t((e=e||self).React)}(this,function(e){\\"use strict\\";e.createElement(\\"h1\\",null)});
//# sourceMappingURL=package-two-umd-with-dep.umd.min.js.map
"
`;

exports[`monorepo umd with dep on other module: package-two-umd-with-dep.umd.min.js.map 1`] = `"{\\"version\\":3,\\"file\\":\\"package-two-umd-with-dep.umd.min.js\\",\\"sources\\":[\\"../src/index.js\\"],\\"sourcesContent\\":[\\"import { createElement } from \\\\\\"react\\\\\\";\\\\n\\\\ncreateElement(\\\\\\"h1\\\\\\", null);\\\\n\\"],\\"names\\":[],\\"mappings\\":\\"sNAEc,KAAM\\"}"`;

exports[`monorepo: package-one.cjs.dev.js 1`] = `
"'use strict';

Object.defineProperty(exports, '__esModule', {
  value: true
});
var index = 1;
exports.default = index;
"
`;

exports[`monorepo: package-one.cjs.js 1`] = `
"'use strict';

if (process.env.NODE_ENV === \\"production\\") {
  module.exports = require(\\"./package-one.cjs.prod.js\\");
} else {
  module.exports = require(\\"./package-one.cjs.dev.js\\");
}
"
`;

exports[`monorepo: package-one.cjs.prod.js 1`] = `
"\\"use strict\\";
Object.defineProperty(exports, \\"__esModule\\", { value: !0 });
var index = 1;
exports.default = index;
"
`;

exports[`monorepo: package-one.esm.js 1`] = `
"var index = 1;

export default index;
"
`;

exports[`monorepo: package-two.cjs.dev.js 1`] = `
"'use strict';

Object.defineProperty(exports, '__esModule', {
  value: true
});
var index = 2;
exports.default = index;
"
`;

exports[`monorepo: package-two.cjs.js 1`] = `
"'use strict';

if (process.env.NODE_ENV === \\"production\\") {
  module.exports = require(\\"./package-two.cjs.prod.js\\");
} else {
  module.exports = require(\\"./package-two.cjs.dev.js\\");
}
"
`;

exports[`monorepo: package-two.cjs.prod.js 1`] = `
"\\"use strict\\";
Object.defineProperty(exports, \\"__esModule\\", { value: !0 });
var index = 2;
exports.default = index;
"
`;

exports[`monorepo: package-two.esm.js 1`] = `
"var index = 2;

export default index;
"
`;

exports[`no module: no-module.cjs.dev.js 1`] = `
"'use strict';

Object.defineProperty(exports, '__esModule', {
  value: true
});
var index = \\"this does not have a module build\\";
exports.default = index;
"
`;

exports[`no module: no-module.cjs.js 1`] = `
"'use strict';

if (process.env.NODE_ENV === \\"production\\") {
  module.exports = require(\\"./no-module.cjs.prod.js\\");
} else {
  module.exports = require(\\"./no-module.cjs.dev.js\\");
}
"
`;

exports[`no module: no-module.cjs.prod.js 1`] = `
"\\"use strict\\";
Object.defineProperty(exports, \\"__esModule\\", { value: !0 });
var index = \\"this does not have a module build\\";
exports.default = index;
"
`;

exports[`object-assign installed: object-assign-installed.cjs.dev.js 1`] = `
"'use strict';

Object.defineProperty(exports, '__esModule', {
  value: true
});

function _interopDefault(ex) {
  return ex && typeof ex === 'object' && 'default' in ex ? ex['default'] : ex;
}

var _objectAssign = _interopDefault(require('object-assign'));

let something = _objectAssign({}, {
  something: true
});

exports.something = something;
"
`;

exports[`object-assign installed: object-assign-installed.cjs.js 1`] = `
"'use strict';

if (process.env.NODE_ENV === \\"production\\") {
  module.exports = require(\\"./object-assign-installed.cjs.prod.js\\");
} else {
  module.exports = require(\\"./object-assign-installed.cjs.dev.js\\");
}
"
`;

exports[`object-assign installed: object-assign-installed.cjs.prod.js 1`] = `
"\\"use strict\\";
function _interopDefault(ex) {
  return ex && \\"object\\" == typeof ex && \\"default\\" in ex ? ex.default : ex;
}
Object.defineProperty(exports, \\"__esModule\\", { value: !0 });
var _objectAssign = _interopDefault(require(\\"object-assign\\"));
let something = _objectAssign({}, { something: !0 });
exports.something = something;
"
`;

exports[`prod checks: prod-checks.cjs.dev.js 1`] = `
"'use strict';

Object.defineProperty(exports, '__esModule', {
  value: true
});

function thing() {
  if (process.env.NODE_ENV !== \\"production\\") {
    return \\"not prod\\";
  }

  return \\"prod\\";
}

exports.thing = thing;
"
`;

exports[`prod checks: prod-checks.cjs.js 1`] = `
"'use strict';

if (process.env.NODE_ENV === \\"production\\") {
  module.exports = require(\\"./prod-checks.cjs.prod.js\\");
} else {
  module.exports = require(\\"./prod-checks.cjs.dev.js\\");
}
"
`;

exports[`prod checks: prod-checks.cjs.prod.js 1`] = `
"\\"use strict\\";
function thing() {
  return \\"prod\\";
}
Object.defineProperty(exports, \\"__esModule\\", { value: !0 }),
  (exports.thing = thing);
"
`;

exports[`umd with dep on other module: umd-with-dep.cjs.dev.js 1`] = `
"'use strict';

var react = require('react');

react.createElement(\\"div\\", null);
"
`;

exports[`umd with dep on other module: umd-with-dep.cjs.js 1`] = `
"'use strict';

if (process.env.NODE_ENV === \\"production\\") {
  module.exports = require(\\"./umd-with-dep.cjs.prod.js\\");
} else {
  module.exports = require(\\"./umd-with-dep.cjs.dev.js\\");
}
"
`;

exports[`umd with dep on other module: umd-with-dep.cjs.prod.js 1`] = `
"\\"use strict\\";
var react = require(\\"react\\");
react.createElement(\\"div\\", null);
"
`;

exports[`umd with dep on other module: umd-with-dep.umd.min.js 1`] = `
"!function(e,t){\\"object\\"==typeof exports&&\\"undefined\\"!=typeof module?t(require(\\"react\\")):\\"function\\"==typeof define&&define.amd?define([\\"react\\"],t):t((e=e||self).React)}(this,function(e){\\"use strict\\";e.createElement(\\"div\\",null)});
//# sourceMappingURL=umd-with-dep.umd.min.js.map
"
`;

exports[`umd with dep on other module: umd-with-dep.umd.min.js.map 1`] = `"{\\"version\\":3,\\"file\\":\\"umd-with-dep.umd.min.js\\",\\"sources\\":[\\"../src/index.js\\"],\\"sourcesContent\\":[\\"import { createElement } from \\\\\\"react\\\\\\";\\\\n\\\\ncreateElement(\\\\\\"div\\\\\\", null);\\\\n\\"],\\"names\\":[],\\"mappings\\":\\"sNAEc,MAAO\\"}"`;
